src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -686,13 +686,13 @@ private function processOnUpdatedHook(object $component, string $frontendName, L
             }
 
             $key = \sprintf('%s.%s', $frontendName, $propName);
-            if (!$dehydratedUpdatedProps->hasPropValue($key)) {
-                continue;
-            }
+            $fullPaths = $dehydratedUpdatedProps->searchFullPathsForProperty($key);
 
-            $this->ensureOnUpdatedMethodExists($component, $funcName);
-            $propertyOldValue = $dehydratedOriginalProps->getPropValue($key);
-            $component->{$funcName}($propertyOldValue);
+            foreach ($fullPaths as $fullPath) {
+                $this->ensureOnUpdatedMethodExists($component, $funcName);
+                $propertyOldValue = $dehydratedOriginalProps->getPropValue($fullPath);
+                $component->{$funcName}($propertyOldValue);
```

Comment:
Here the receiving callable as no way to get what is "old value". If a "*" is used, we should pass the full prop